modThread/modThread.py

```diff
@@ -4,280 +4,404 @@
 
-    ### End Mod Category
+    @modthread.command(name="settings")
+    async def settings(
+        self,
+        ctx: commands.Context
+    ) -> None:
+        """Settings command group"""
 
-    # endregion
+        primary_category = await self._get_primary_category(ctx.guild)
+        management_role = await self._get_management_role(ctx.guild)
 
-    # region json db
-    async def _primary_category(self, ctx):
-        return ctx.guild.get_channel(
-            await self.config.guild(ctx.guild).PrimaryCategory()
+        settings_embed = discord.Embed(
+            title="ModThread Settings",
+            description="Current configuration for modThread Cog.",
+            color=discord.Color.blue(),
         )
 
-    async def _save_primary_category(self, ctx, primary_category):
-        await self.config.guild(ctx.guild).PrimaryCategory.set(primary_category)
-
-    async def _rules_category(self, ctx):
-        return ctx.guild.get_channel(await self.config.guild(ctx.guild).RulesCategory())
-
-    async def _save_rules_category(self, ctx, rules_category):
-        await self.config.guild(ctx.guild).RulesCategory.set(rules_category)
-
-    async def _numbers_category(self, ctx):
-        return ctx.guild.get_channel(
-            await self.config.guild(ctx.guild).NumbersCategory()
-        )
+        # Check channel values before mention to avoid exception
+        if primary_category:
+            settings_embed.add_field(
+                name="Primary Category",
+                value=primary_category.jump_url,
+                inline=False
+            )
+        else:
+            settings_embed.add_field(
+                name="Primary Category", value="Not Set", inline=False
+            )
 
-    async def _save_numbers_category(self, ctx, numbers_category):
-        await self.config.guild(ctx.guild).NumbersCategory.set(numbers_category)
+        if management_role:
+            settings_embed.add_field(
+                name="Management Role",
+                value=management_role.mention,
+                inline=False
+            )
+        else:
+            settings_embed.add_field(
+                name="Management Role", value="Not Set", inline=False
+            )
 
-    async def _mods_category(self, ctx):
-        return ctx.guild.get_channel(await self.config.guild(ctx.guild).ModsCategory())
+        await ctx.send(embed=settings_embed)
+
+    @modthread.command(name="groups", alias="group")
+    async def groups(
+        self,
+        ctx: commands.Context,
+        action: str | None,
+        group: str | None,
+        category: discord.CategoryChannel | None,
+        role: discord.Role | None
+    ):
+        """Groups command group"""
+
+        if action:
+            action = action.lower()
+            if action in ["add", "update"]:
+                group = group.lower()
+                await self._set_group(ctx.guild, group, category, role)
+                await ctx.send('Ok!')
```

Comment:
Missing a return here. The help message is displayed after adding or updating a new group

Reply:
I want it to display the help message, because it contains a list of the current groups.
![image](https://github.com/RSC-NA/bot-RSCBot/assets/124588/b0e12b4a-92e6-4446-8632-3e07d877ed3f)

Reply:
See "Defined Groups" at the bottom of the list.

Perhaps I could remove the Syntax/actions stuff... :shrug: I liked it though.

Reply:
I removed the syntax/actions if you successfully add/delete a group.